website/src/components/DocSearch/index.js

```diff
@@ -0,0 +1,46 @@
+import React, { useRef } from 'react';
+import { useHistory } from '@docusaurus/router';
+import docsearch from 'docsearch.js';
+
+/**
+ * Init Docsearch on an input (children)
+ */
+const DocSearch = ({ searchElId, children, algoliaOptions }) => {
+  const initialized = useRef(false);
```

Comment:
I realize this is a way to do state without running side effects, but isn't that already covered by the dependency list in the `useEffect`? Also, if the intention is to only run the side effect once during initialization, then passing `[]` as the side effect dependency seems more idiomatic. what do you think?

Reply:
Nice catch! `initialized` came from original component before "swizzling" 
I will fix this.

Intention is to initialise the search on mount but also if the children change
There is a, somewhat fragile, link because the `children` and the `searchElId`
Algolia will init the search on a child element of `searchElId`
So if the `children`  change, the `searchElId` should be changed too and the search function re-initialised.

Reply:
if you feel like it, making this a `useAlgolia` hook which receives an `inputRef` and then observing `inputRef.current` to grab the element id is another way to go about this. That way the id is only set once in the input ;)

Reply:
@hlminh2000 changed to use algolia hook  :)